src/extensions/lens-main-extension.ts

```diff
@@ -2,14 +2,18 @@ import type { MenuRegistration } from "./registries/menu-registry";
 import { observable } from "mobx";
 import { LensExtension } from "./lens-extension"
 import { WindowManager } from "../main/window-manager";
-import { getPageUrl } from "./registries/page-registry"
+import { getExtensionPageUrl } from "./registries/page-registry"
 
 export class LensMainExtension extends LensExtension {
   @observable.shallow appMenus: MenuRegistration[] = []
 
-  async navigate(location?: string, frameId?: number) {
+  async navigate<P extends object>(pageId?: string, params?: P, frameId?: number) {
```

Comment:
nice. Would we introduce another function for navigating to a page in another extension? (adding another optional param here for the extension id would probably be too unwieldy) Like
```
async navigateExtension<P extends object>(extensionId: extId, pageId?: string, params?: P, frameId?: number)
```

Reply:
This breaks the API compatibility, right? Would it be too bad if we kept `navigation` as is and introduce this as a new method (with additional `extensionId?`)?

Reply:
It breaks API only in second argument, right? It's done to be more aligned with `ext.navigate` for renderer process.
And using this `frameId` mostly for internal usage, at least for now?
Otherwise we can introduce new method like `ext.navigateWith(opts: PageMenuTarget & {frameId?: number})`. 
Wdyt?

Reply:
@ixrock you mean that `this.navigate("/my-page")` still works?

Reply:
@jakolehm Yes, it should